From 344bba7458d52e0e63875e36b0419bc17ab28572 Mon Sep 17 00:00:00 2001
From: Artur Wieczorek <artwik@wp.pl>
Date: Sat, 21 Mar 2015 23:51:57 +0100
Subject: [PATCH] When wxUSE_ACTIVITYINDICATOR is enabled check if
 wxGraphicsContext is also enabled.

wxActivityIndicatorImpl requires wxGraphicsContext to compile successfully (it is referred e.g. in wxActivityIndicatorImpl::OnPaint).Additional check of setup would prevent occurring unexpected compilation errors.
---
 include/wx/chkconf.h | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/include/wx/chkconf.h b/include/wx/chkconf.h
index c67c67c..ff49eac 100644
--- a/include/wx/chkconf.h
+++ b/include/wx/chkconf.h
@@ -2247,6 +2247,15 @@
 #   endif
 #endif /* wxUSE_PREFERENCES_EDITOR */
 
+#if wxUSE_ACTIVITYINDICATOR && !wxUSE_GRAPHICS_CONTEXT
+#   ifdef wxABORT_ON_CONFIG_ERROR
+#       error "wxUSE_ACTIVITYINDICATOR requires wxGraphicsContext"
+#   else
+#       undef wxUSE_ACTIVITYINDICATOR
+#       define wxUSE_ACTIVITYINDICATOR 0
+#   endif
+#endif /* wxUSE_ACTIVITYINDICATOR */
+
 #endif /* wxUSE_GUI */
 
 #endif /* _WX_CHKCONF_H_ */
-- 
1.9.5.msysgit.0

